static/js/services/delete-docs.js

```diff
@@ -11,26 +12,35 @@ var _ = require('underscore');
       $log,
       $q,
       DB,
-      ExtractLineage
+      ExtractLineage,
+      Session
     ) {
 
       'ngInject';
 
       var getParent = function(doc) {
         if (doc.type === 'person' && doc.parent && doc.parent._id) {
-          return DB()
-            .get(doc.parent._id)
+          return DB().get(doc.parent._id);
+        }
+        return $q.resolve();
+      };
+
+      var getParentsToUpdate = function(docs) {
+        return $q.all(docs.map(function(doc) {
+          return getParent(doc)
             .then(function(parent) {
-              if (parent.contact &&
-                  parent.contact._id &&
-                  parent.contact._id === doc._id) {
-                // this doc is the contact for the parent - update parent
+              var shouldUpdateParentContact = parent && parent.contact && parent.contact._id && parent.contact._id === doc._id;
+              if (shouldUpdateParentContact) {
                 parent.contact = null;
                 return parent;
               }
             });
-        }
-        return $q.resolve();
+          }))
+          .then(function(parents) {
+            return parents.filter(function(parent) {
+              return typeof parent !== 'undefined';
```

Comment:
FWIW `return parent` is legit, I don't think you have to be that explicit?

Reply:
Yeah `return parent` should suffice 👍